src/probe/probe.c

```diff
@@ -1041,10 +1088,14 @@ int probe_point_add(uint32_t count, struct probe_point *probe)
 		_probe->probe_points[first_free].stream_tag =
 			probe[i].stream_tag;
 
-		notifier_register(_probe, dev->cb, NOTIFIER_ID_BUFFER_PRODUCE,
-				  &probe_cb_produce, 0);
-		notifier_register(_probe, dev->cb, NOTIFIER_ID_BUFFER_FREE,
-				  &probe_cb_free, 0);
+		if (probe[i].purpose == PROBE_PURPOSE_EXTRACTION) {
```

Comment:
Should this be `if (probe[i].purpose != PROBE_PURPOSE_LOGGING)` instead?

Reply:
Oh yes, buffer-id-produce notifier is also used for injection, will update.